cpt/runner.py

```diff
@@ -112,6 +116,11 @@ def run(self):
                                 else:
                                     self.printer.print_message("Skipping upload for %s, "
                                                                "it hasn't been built" % package_id)
+                            elif (installed["recipe"]["id"] in self._upload_dependencies) or \
+                                 ("all" in self._upload_dependencies):
+                                reference = installed["recipe"]["id"]
+                                package_id = installed['packages'][0]['id']
```

Comment:
This line will fail if a dependency has no packages (e.g. msys2_installer/latest@bincrafters/stable)

Reply:
Done, now only built packages will be uploaded.

Reply:
But there is still no check if `installed['packages']` has no packages, right?

Reply:
Done!